compiler/rustc_resolve/src/lib.rs

```diff
@@ -580,6 +584,9 @@ struct ModuleData<'ra> {
     /// Used to memoize the traits in this module for faster searches through all traits in scope.
     traits: RefCell<Option<Box<[(Ident, NameBinding<'ra>)]>>>,
 
+    /// In case supertraits are relevant in the module, the DefIds are collected, too.
+    supertraits: RefCell<IndexSet<DefId>>,
```

Comment:
I don't think this is necessary.
Since `use ChildTrait::ItemFromParent;` and similar do not work and are not supposed to work, the whole implementation can be localized to `compiler\rustc_resolve\src\late.rs`.

Perhaps as two passes in late resolver.
Since if the first (regular) pass we do not always have `trait Parent` late-resolved when late-resolving
```rust
trait Child: Parent { ... }
```
, but we may need `trait Parent` late-resolved here.
So maybe the first pass can just keep a hole and some bookkeeping, so that the hole can be filled in a second pass.